homework/hackyourtemperature/app.js

```
@@ -0,0 +1,45 @@
+import express from "express";
+import { API_KEY } from "./sources/keys.js";
+import fetch from "node-fetch";
+
+const app = express();
+
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send("hello from backend to frontend!");
+});
+
+app.post("/weather", async (req, res) => {
+  try {
+    const { cityName } = req.body;
+
+    if (!cityName) {
+      throw new Error("City name is missing");
+    }
+
+    const response = await fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&APPID=${API_KEY}`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    const data = await response.json();
+
+    if (data.cod === "404") {
+      res.json({ weatherText: "City is not found!" });
+    } else {
+      const cityName = data.name;
+      const temperature = data.main.temp;
+      const weatherText = `Temperature in ${cityName}: ${temperature}°K`;
+      res.json({ weatherText });
+    }
+  } catch (error) {
+    res.status(400).send(`Error: ${error.message}`);
+  }
```

Here you can actually return an error with a status code of 500 because if an error happens its very likely because of the weather API failing to respond back. Afterall, sometimes even if your request to that API is correct and the cityName is fine, the external API can still fail on their side. So, you are handling it correct but you are returning the wrong status code and the end user will think its because of them and not because of the external API.

---

thank you very much , I will fix it